src/main/scala/searchengine/processors/WordOccurence.scala

```diff
@@ -0,0 +1,28 @@
+package searchengine.processors
+
+import searchengine.Processor
+import searchengine.http.HttpResponse
+import searchengine.processors.WordOccurence.{Link, OccurrenceCount, Word}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+case class WordOccurence(linkWordOccurenceMap: Map[Link, Map[Word, OccurrenceCount]])
+
+object WordOccurence {
+  type Link = String
+  type OccurrenceCount = Int
+  type Word = String
+
+  def wordsOf(text: String): List[String] = text.split("\\W+").toList.filter(_.nonEmpty)
+}
+
+object WordOccurencerCounter extends Processor[WordOccurence] {
+  def apply(url: String, response: HttpResponse): Future[WordOccurence] = Future {
+    if (response.isSuccess && (response.isHTMLResource || response.isPlainTextResource))
+      WordOccurence(Map(url -> Await.result(WordCounter(url, response).map(_.wordToCount), Duration.Inf)))
```

Comment:
Why do we need to await the result just to pass it into another future? Cannot we use flatMap?

Below is my example, not sure if it will compile though :D
```scala
WordCounter(url, response).map(_.wordToCount).flatMap(count => WordOccurence(Map(url-> count)))
```